eng/pipelines/tools-repo-versioning.yml

```diff
@@ -0,0 +1,42 @@
+# Tag the tools repos (azure-sdk-tools or azure-sdk-build-tools) to create some form of versioning.
+resources:
+  repositories:
+    - repository: azure-sdk-tools
+      type: github
+      name: Azure/azure-sdk-tools
+      endpoint: azure
+    - repository: azure-sdk-build-tools
+      type: git
+      name: internal/azure-sdk-build-tools
+
+trigger: none
+
+pr: none
+
+pool:
+  vmImage: windows-2019
+
+jobs:
+  - job: TagToolsRepo
+    displayName: Version Repo with Git Tag
+
+    steps:
+    - checkout: azure-sdk-tools
```

Comment:
This limits this yml file to only ever working for these 2 repos. If we really want this to be flexible then we should simply pass in the repository and the branch or sha. However by doing that we lose some flexibility. In particular we lose the ability to have this automatically trigger. Personally I kind of likely having one pipeline per repo that handles this. Of course we can use the same yml (or template) in each and configure them as such but having a specific pipeline for each seems like it would be better.